public/app/features/explore/state/actions.ts

```diff
@@ -530,6 +533,124 @@ export const runQueries = (exploreId: ExploreId): ThunkResult<void> => {
+      return;
+    }
+
+    // Some datasource's query builders allow per-query interval limits,
+    // but we're using the datasource interval limit for now
+    const minInterval = datasourceInstance?.interval;
+
+    stopQueryState(querySubscription);
+
+    const datasourceId = datasourceInstance?.meta?.id;
+
+    const queryOptions: QueryOptions = {
+      minInterval: minInterval ?? '1s',
+      // maxDataPoints is used in:
+      // Loki - used for logs streaming for buffer size, with undefined it falls back to datasource config if it supports that.
+      // Elastic - limits the number of datapoints for the counts query and for logs it has hardcoded limit.
+      // Influx - used to correctly display logs in graph
+      maxDataPoints: mode === ExploreMode.Logs && datasourceId === 'loki' ? undefined : containerWidth,
+      liveStreaming: live,
+      showingGraph,
+      showingTable,
+      mode,
+    };
+
+    const transaction = buildQueryTransaction(queries, queryOptions, newTimeRange, !!scanning);
+
+    let firstResponse = true;
+    dispatch(changeLoadingStateAction({ exploreId, loadingState: LoadingState.Loading }));
+
+    const newQuerySub = runRequest(datasourceInstance, transaction.request)
+      .pipe(
+        // Simple throttle for live tailing, in case of > 1000 rows per interval we spend about 200ms on processing and
+        // rendering. In case this is optimized this can be tweaked, but also it should be only as fast as user
+        // actually can see what is happening.
+        live ? throttleTime(500) : identity,
+        map((data: PanelData) => preProcessAppendPanelData(data, queryResponse, direction))
+      )
+      .subscribe((data: PanelData) => {
+        if (!data.error && firstResponse) {
+          // We save queries to the URL here so that only successfully run queries change the URL.
+          dispatch(stateSave());
+        }
+
+        firstResponse = false;
+        const dataAfterDrop = {
+          ...data,
+          series:
+            direction === QueryDirection.backward ? data.series.slice(1) : data.series.slice(0, data.series.length - 1),
+        };
+        dispatch(queryStreamUpdatedAction({ exploreId, response: dataAfterDrop }));
+
+        // Keep scanning for results if this was the last scanning transaction
+        if (getState().explore[exploreId].scanning) {
+          if (data.state === LoadingState.Done && data.series.length === 0) {
+            const range = getShiftedTimeRange(-1, getState().explore[exploreId].range);
+            dispatch(updateTime({ exploreId, absoluteRange: range }));
+            dispatch(runQueries(exploreId));
+          } else {
+            // We can stop scanning if we have a result
+            dispatch(scanStopAction({ exploreId }));
+          }
+        }
+      });
+
+    dispatch(queryStoreSubscriptionAction({ exploreId, querySubscription: newQuerySub }));
+  };
+};
+
```

Comment:
Similar to `runQueries`, but runs append queries and uses slightly different data and time range preprocessing.

Reply:
In general this should be refactored as it basically the same as `runQueries`. Either make some of the behaviour plug-able in `runQuery` or it could be split and reused.